src/ellipse.rs

```diff
@@ -229,15 +229,34 @@ impl Shape for Ellipse {
         PI * x * y
     }
 
+    /// Approximate the ellipse perimeter.
+    ///
+    /// This uses a numerical approximation. The absolute error between the calculated perimeter
+    /// and the true perimeter is bounded by `accuracy` (modulo floating point rounding errors).
+    ///
+    /// For circular ellipses (equal horizontal and vertical radii), the calculated perimeter is
+    /// exact.
     #[inline]
     fn perimeter(&self, accuracy: f64) -> f64 {
-        // TODO rather than delegate to the bezier path, it is possible to use various series
-        // expansions to compute the perimeter to any accuracy. I believe Ramanujan authored the
-        // quickest to converge. See
-        // https://www.mathematica-journal.com/2009/11/23/on-the-perimeter-of-an-ellipse/
-        // and https://en.wikipedia.org/wiki/Ellipse#Circumference
-        //
-        self.path_segments(0.1).perimeter(accuracy)
+        let radii = self.radii();
+
+        if radii.is_nan() {
```

Comment:
Maybe change this to `!is_finite()`?

Reply:
Hmm... my intention was to return `f64::INFINITY` in case of infinite radii. I'm not sure if that's desirable though, and I think it broke somewhere along the way. Didn't add a test for it early on.

Reply:
I remember now: because ellipses are defined by an `Affine`, constructing an ellipse with infinite radii causes the ellipse radii to be calculated as NaN. I have changed the check to `!is_finite()` as you suggested, to make it robust if the inner definition of an ellipse were ever to change.